test-apps/appui-test-app/appui-test-providers/src/ui/dialogs/ISVPIPView.tsx

```diff
@@ -0,0 +1,188 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+
+import * as React from "react";
+import { IModelApp, ScreenViewport, ViewManip, ViewState } from "@itwin/core-frontend";
+import { AbstractMenuItemProps } from "@itwin/appui-abstract";
+import { ContentControl, ContentViewManager, FloatingViewportContent, FloatingViewportContentControl, FrontstageManager, UiFramework, useActiveIModelConnection } from "@itwin/appui-react";
+
+import "./ISVPIPView.scss";
+import { Id64String } from "@itwin/core-bentley";
+import ViewDefinitionSelector, { getViewDefinitions } from "../components/ViewDefinitionSelector";
+import { ViewIdChangedEventArgs, ViewportComponentEvents } from "@itwin/imodel-components-react";
+import { Presentation, TRANSIENT_ELEMENT_CLASSNAME } from "@itwin/presentation-frontend";
+import { KeySet } from "@itwin/presentation-common";
+interface ISVPIPViewDefInterfaceLocal {
+  id: string; class: string; label: string;
+}
+export function ISVPIPView({ contentId, showViewPicker }: { contentId: string, showViewPicker?: boolean }) {
+  const getIds=(args: Readonly<KeySet>) =>{
+    let allIds: Set<string> = new Set<string>();
+    args.instanceKeys.forEach((ids: Set<string>, key: string) => {
+    // Avoid transient elements
+      if (key !== TRANSIENT_ELEMENT_CLASSNAME) allIds = new Set([...allIds, ...ids]);
+    });
+    return allIds;
+  };
+  const menuItems: AbstractMenuItemProps[] = React.useMemo(() => {
+    return [
+      {
+        id: "Item1", label: "Item ~1", icon: "icon-placeholder",
+        submenu: [
+          { id: "0", item: { label: "SubMenu Item ~1", icon: "icon-placeholder", execute: () => { } } },
+          { id: "1", item: { label: "SubMenu Item ~2", icon: "icon-placeholder", execute: () => { } } },
+        ],
+      },
+      {
+        id: "Item2", item: { label: "Item ~2", icon: "icon-placeholder", execute: () => { } },
+      },
+      {
+        id: "Item3", item: { label: "Item ~3", icon: "icon-placeholder", execute: () => { } },
+      },
+    ];
+  }, []);
+
+  const activeIModelConnection = useActiveIModelConnection();
+  const divRef = React.useRef<HTMLDivElement>(null);
+
+  const [initialViewState, setInitialViewState] = React.useState<ViewState | undefined>(undefined);
+  const [twoDViewDefinitions, settwoDViewDefinitions] = React.useState<ISVPIPViewDefInterfaceLocal[]>([]);
+  const [threeDViewDefinitions, setthreeDViewDefinitions] = React.useState<ISVPIPViewDefInterfaceLocal[]>([]);
+
+  const handleViewIdChange = React.useCallback (async (args: ViewIdChangedEventArgs) => {
```

Comment:
This callback is marked async but we do not use `await` or return a promise, from what I see, UiEvent -> BeUiEvent -> BeEvent are not really handling promises anyway, so this is giving a false impression of "asynchronicity". I'm not sure what is the intention here.

Reply:
That's a mistake. I removed the async from it.